lib/persistence.js

```diff
@@ -107,7 +107,7 @@ class Persistence {
       if (fieldName !== '_id') { // The special _id index is managed by datastore.js, the others need to be persisted
         lines.push(this.afterSerialization(model.serialize({
           $$indexCreated: {
-            fieldName: fieldName,
+            fieldName: this.db.indexes[fieldName].fieldName,
```

Comment:
When fixing comment https://github.com/seald/nedb/pull/27#issuecomment-1321231668, this line will need to be fixed as well